src/hooks/useUniversalRouter.ts

```diff
@@ -21,10 +22,16 @@ interface SwapOptions {
   feeOptions?: FeeOptions
 }
 
+/**
+ * Returns a callback to submit a transaction to the universal router.
+ *
+ * The callback returns the TransactionResponse if the transaction was submitted,
+ * or undefined if the user rejected the transaction.
+ **/
 export function useUniversalRouterSwapCallback(trade: InterfaceTrade | undefined, options: SwapOptions) {
   const { account, chainId, provider } = useWeb3React()
 
-  return useCallback(async (): Promise<TransactionResponse> => {
+  return useCallback(async (): Promise<TransactionResponse | undefined> => {
```

Comment:
nit: I think it would be clearer if you returned something concrete, not `undefined`. Could you create a `Symbol('Rejected transaction')` and return that instead? Or even just return `false` or `null`?